src/map/skill.cpp

```diff
@@ -15672,18 +15672,26 @@ bool skill_check_condition_castbegin(struct map_session_data* sd, uint16 skill_i
 			case RA_AIMEDBOLT:
 				break;
 			default:
-				if (require.weapon&(1<<W_REVOLVER))
-					clif_msg(sd, SKILL_NEED_REVOLVER);
-				else if (require.weapon&(1<<W_RIFLE))
-					clif_msg(sd, SKILL_NEED_RIFLE);
-				else if (require.weapon&(1<<W_GATLING))
-					clif_msg(sd, SKILL_NEED_GATLING);
-				else if (require.weapon&(1<<W_SHOTGUN))
-					clif_msg(sd, SKILL_NEED_SHOTGUN);
-				else if (require.weapon&(1<<W_GRENADE))
-					clif_msg(sd, SKILL_NEED_GRENADE);
-				else
-					clif_skill_fail(sd, skill_id, USESKILL_FAIL_THIS_WEAPON, 0);
+				switch((unsigned int)log2(require.weapon)) {
```

Comment:
Not sure if this is the best way of reversing the bitwise, but at least it works.

Reply:
I guess in this case it would have been best to exchange the bitwise and with an equality operator.

Since log2 is C99 and we are only featuring VS2013 and above we should be fine with your approach as well though 👍

Reply:
The only way I can think of is to do like @Lemongrass3110 suggested and have specific equality checks. Maybe just leave a comment and say what the `log2` function is used here for if we keep it this way. :)